src/audio/google/google_rtc_audio_processing.c

```diff
@@ -855,26 +902,73 @@ static int google_rtc_audio_processing_process(struct processing_module *mod,
 			      (const void **)&src_buf_start,  &src_buf_size);
 	assert(!ret);
 	src_buf_end = src_buf_start + src_buf_size;
-
+	/* The second optimization eliminates the inner loop
+	 * and replaces it with pointer arithmetic for speedier access.
+	 * To reduce cycle waste, the data copy check is moved outside of the loop.
+	 */
+	/* Initialize error_code to 0 (no error) */
+	int error_code = 0;
 	buffer_offset = 0;
-	for (int i = 0; i < cd->num_frames; i++) {
-		for (channel = 0; channel < cd->num_capture_channels; channel++)
+	const int16_t *src_end = src + cd->num_frames * cd->config.output_fmt.channels_count;
+
+	/* Check if the calculated end of the source buffer exceeds the actual end of the buffer */
+	if (src_end >= (const int16_t *)src_buf_end)
```

Comment:
There's quite many of these pointer check and wrap code lines. An inline function of macro (ternary?) would make cleaner looking code.

Reply:
E.g. cir_buf_wrap() from audio_stream.h.